webnode/src/redux/epics/treasure-hunt-epics.js

```diff
@@ -94,43 +92,52 @@ const findTreasureEpic = (action$, store) => {
     .mergeMap(action => {
       const { dataMapHash, chunkIdx } = action.payload;
 
-      const address = dataMapHash;
+      const address = iota.toAddress(
+        iota.utils.toTrytes(Encryption.obfuscate(dataMapHash))
+      );
       //    const address =
       //      "HT9MZQXKVBVT9AYVTISCLELYWXTILJDIMHFQRGS9YIJUIRSSNRZFIZCHYHQHKZIPGYYCSUSARFNSXD9UY";
 
       return Observable.fromPromise(
         iota.findMostRecentTransaction(address)
       ).mergeMap(transaction => {
         const message = iota.parseMessage(transaction.signatureMessageFragment);
-        const sideChain = Sidechain.generate(address);
+        const sideChain = Sidechain.generate(dataMapHash);
         store.dispatch({
           //TODO Remove this dispatch
           type: "IOTA_RETURN"
         });
 
-        const chainWithTreasure = _.find(
-          sideChain,
-          hashedAddress => !!Encryption.decrypt(message, hashedAddress)
-        );
+        const chainWithTreasure = _.find(sideChain, hashedAddress => {
+          return !!Encryption.decryptTreasure(
+            hashedAddress,
+            transaction.signatureMessageFragment,
+            dataMapHash
+          );
+        });
 
         const [_obfHash, nextDataMapHash] = Encryption.hashChain(dataMapHash);
 
-        return Observable.if(
```

Comment:
I wasn't able to figure out why, but this Observable.if was still running the first observable, even when !!chainWithTreasure was false.  So for the time being I've just wrapped those things in an if/else.

Reply:
Small nit: It would be cleaner to put the ternary inside the the `Observable.of()` function since an observable must always be returned.

```js
return Observable.of(!!chainWIthTreasure ? ... : ...);
```

Reply:
Okay I'll change it